pyro/distributions/torch_patch.py

```diff
@@ -59,6 +59,16 @@ def _torch_linspace(*args, **kwargs):
     return ret
 
 
+# Fixes a shape error in Multinomial.support with inhomogeneous .total_count
+@patch_dependency('torch.distributions.Multinomial.support')
+@torch.distributions.constraints.dependent_property
+def _Multinomial_support(self):
```

Comment:
Nice catch, I completely forgot the Multinomial upstream only supports integer total count.